extern/boostd-data/yugabyte/service.go

```diff
@@ -537,28 +537,54 @@ func (s *Store) addMultihashesToPieces(ctx context.Context, pieceCid cid.Cid, re
 	insertPieceOffsetsQry := `INSERT INTO PayloadToPieces (PayloadMultihash, PieceCid) VALUES (?, ?)`
 	pieceCidBytes := pieceCid.Bytes()
 
-	var batch *gocql.Batch
-	for allIdx, rec := range recs {
-		if batch == nil {
-			batch = s.session.NewBatch(gocql.UnloggedBatch).WithContext(ctx)
-			batch.Entries = make([]gocql.BatchEntry, 0, s.settings.InsertBatchSize)
+	threadBatch := len(recs) / 32 // split the slice into go-routine batches for ~32 workers
```

Comment:
Shall we take an opportunity and extract batching / parallelisation logic into a separate function as it used twice

Reply:
Otherwise lgtm

Reply:
I am merging this as is for now. We can shorten the code for next release.